src/leafletwidget/widget.ts

```diff
@@ -47,6 +51,25 @@ class MapWidget extends Widget {
         max_zoom : 18,
     }).addTo(this._map);
 
+    // Since we keep track of the widget size, we monkeypatch the map
+    // to use our information instead of doing a DOM read every time it needs
+    // the size info. We can stop monkeypatching when we have a size hint change
+    // available from leaflet (cf.
+    // https://github.com/Leaflet/Leaflet/issues/4200#issuecomment-233616337 and
+    // https://github.com/jupyter/jupyterlab/pull/454#discussion_r71349224)
+    this._map.getSize = () => {
+      let map: any = this._map;
+      if (!map._size || map._sizeChanged) {
+        if (this._width < 0 || this._height < 0) {
+          return map.prototype.getSize.call(map);
+        } else {
```

Comment:
as a style issue, to emphasize the return? I saw it as a style issue to emphasize the binary nature of the two branches. But I'm happy to defer - no strong opinions here.